pkg/commands/kvstoreentry/create.go

```diff
@@ -26,20 +28,233 @@ func NewCreateCommand(parent cmd.Registerer, g *global.Data, m manifest.Data) *C
-	c.CmdClause.Flag("value", "Value").Required().StringVar(&c.Input.Value)
+	c.CmdClause.Flag("value", "Value").StringVar(&c.Input.Value)
 	return &c
 }
 
+// CreateCommand calls the Fastly API to insert a key into an kv store.
+type CreateCommand struct {
+	cmd.Base
+	dirConcurrency int
+	dirPath        string
+	filePath       string
+	manifest       manifest.Data
+	stdin          bool
+
+	Input fastly.InsertKVStoreKeyInput
+}
+
 // Exec invokes the application logic for the command.
-func (c *CreateCommand) Exec(_ io.Reader, out io.Writer) error {
+func (c *CreateCommand) Exec(in io.Reader, out io.Writer) error {
+	if err := c.CheckFlags(); err != nil {
+		return err
+	}
+
+	if c.stdin {
+		return c.ProcessStdin(in, out)
+	}
+
+	if c.filePath != "" {
+		return c.ProcessFile(out)
+	}
+
+	if c.dirPath != "" {
+		return c.ProcessDir(out)
+	}
+
+	if c.Input.Key == "" || c.Input.Value == "" {
+		return fsterr.ErrInvalidKVCombo
+	}
+
 	err := c.Globals.APIClient.InsertKVStoreKey(&c.Input)
 	if err != nil {
 		c.Globals.ErrLog.Add(err)
 		return err
 	}
 
-	text.Success(out, "Inserted key %s into kv store %s", c.Input.Key, c.Input.ID)
+	text.Success(out, "Inserted key %s into KV Store %s", c.Input.Key, c.Input.ID)
+
+	return nil
+}
+
+// CheckFlags ensures only one of the three specified flags are provided.
+func (c *CreateCommand) CheckFlags() error {
+	flagCount := 0
+	if c.stdin {
+		flagCount++
+	}
+	if c.filePath != "" {
+		flagCount++
+	}
+	if c.dirPath != "" {
+		flagCount++
+	}
+	if flagCount > 1 {
+		return fsterr.ErrInvalidStdinFileDirCombo
+	}
 	return nil
 }
+
+// ProcessStdin streams STDIN to the batch API endpoint.
+func (c *CreateCommand) ProcessStdin(in io.Reader, out io.Writer) error {
+	// Determine if 'in' has data available.
+	if in == nil || text.IsTTY(in) {
+		return fsterr.ErrNoSTDINData
+	}
+	return c.CallBatchEndpoint(in, out)
+}
+
+// ProcessFile streams a JSON file content to the batch API endpoint.
+func (c *CreateCommand) ProcessFile(out io.Writer) error {
+	f, err := os.Open(c.filePath)
+	if err != nil {
+		c.Globals.ErrLog.Add(err)
+		return err
+	}
+	return c.CallBatchEndpoint(f, out)
+}
+
+// ProcessDir concurrently reads files from the given directory structure and
+// uploads each file to the set-value-for-key endpoint where the filename is the
+// key and the file content is the value.
+//
+// NOTE: Unlike ProcessStdin/ProcessFile content doesn't need to be base64.
+func (c *CreateCommand) ProcessDir(out io.Writer) error {
+	path, err := filepath.Abs(c.dirPath)
+	if err != nil {
+		return err
+	}
+
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	spinner, err := text.NewSpinner(out)
+	if err != nil {
+		return err
+	}
+
+	err = spinner.Start()
+	if err != nil {
+		return err
+	}
+	fileLength := len(files)
+	msg := "Processed %d of %d files"
+	spinner.Message(fmt.Sprintf(msg, 0, fileLength) + "...")
+
+	base := filepath.Base(path)
+	errors := []ProcessErr{}
```

Comment:
I typically see this as `var errors []ProcessErr`

Reply:
Fixed in [4f74c01](https://github.com/fastly/cli/pull/927/commits/4f74c01f64974c1cd398af98a0a1c07a080fab76)